File: 2.js
var app = (function(){
 	function fib(n){
 	 if(n <= 1)return 1; 
  	return fib(n-1) + fib(n-2);

}

function isDivisible(n, div){
	return n % div == 0;
}

function evenFibonnacciNumbers(){
	var result = 0;
	for(var i = 0; i < 100000; i++){
		var num = fib(i);
       		if(num > 4000000)return result;
        	if(isDivisible(num, 2 )){
        		 result  += num;
                } 
        }
}

return {
  evenFibonnacciNumbers: evenFibonnacciNumbers
}
}());
console.log(app.evenFibonnacciNumbers());

File: ruby/2.rb



def fibonacci(n)
    if n <= 1 
        return 1 
    end
    return fibonacci(n-1) + fibonacci(n-2)
end

result = 0
LIMIT = 4000000
(1..10000).each do |val|
    fib  = fibonacci(val)
    if fib > LIMIT 
        break 
    end
    if ( fib % 2  == 0 ) 
        result += fib
    end
end

p result